### PR TITLE
urdfdom: 3.3.1 -> 4.0.0

### PR DESCRIPTION
**Pull request by lopsided98**
## Description of changes

Update urdfdom and urdfdom-headers. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 282671` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.crocoddyl</li>
  </ul>
</details>
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>crocoddyl</li>
    <li>pinocchio</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.pinocchio</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.pinocchio</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>urdfdom</li>
    <li>urdfdom-headers</li>
  </ul>
</details>

**Comment by nim65s** [comment]
I'll try to fix those

**Comment by nim65s** [comment]
Result of `nixpkgs-review pr 282671` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.crocoddyl</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>crocoddyl</li>
    <li>pinocchio</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.pinocchio</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.pinocchio</li>
    <li>urdfdom</li>
    <li>urdfdom-headers</li>
  </ul>
</details>

**Review by nim65s — APPROVED** [review]
adding pull request numbers would help identify and track the commit

File: pkgs/development/libraries/urdfdom-headers/default.nix [hunk 1]
```diff
@@ -2,20 +2,20 @@
...
-    hash = "sha256-ry5wDMRxR7TtupUghe9t1XP0XMvWKiOesO5RFHPrSdI=";
+    hash = "sha256-FQSnYuTc40MOxyFsMPfoCIonP+4AUQxdq74eoQ9tOoo=";
   };
 
   patches = [
     # Fix CMake relative install dir assumptions (https://github.com/ros/urdfdom_headers/pull/66)
     (fetchpatch {
-      url = "https://github.com/ros/urdfdom_headers/commit/c9c993147bbf18d5ec83bae684c5780281e529b4.patch";
-      hash = "sha256-BnYPdcetYSim2O1R38N0d1tY0Id++AgKNic8+dlM6Vg=";
+      url = "https://github.com/ros/urdfdom_headers/commit/6e0cea148c3a7123f8367cd48d5709a4490c32f1.patch";
```
[the remark this answers is not in the document]
Comment (lopsided98) [reply pkgs/development/libraries/urdfdom-headers/default.nix:17]: The link is in the comment, I don't see why it needs to be in the URL.

**Comment by nim65s** [comment]
good bot :)
I'll check what is happening with pinocchio.

**Comment by nim65s** [comment]
I pushed a fix in #318625, not sure how best to handle the PR in this case.

**Pull request closed without merging**